### ISSUE TITLE
Going back to device locale

### ISSUE THREAD

**Issue by amatveyakin** [aissat/easy_localization#206]
*TL;DR.* Does easy_localization provide a way to select the locale that the app would've started with assuming `startLocale` were `null`?

*Details.* My app allows users to choose a language explicitly. One of the options is “System default”. I'm stuck trying to figure out how to go back to the default locale. I thought something like this would work:

```dart
context.locale = language != null
    ? Locale(language)
    : localeFromString(await findSystemLocale());
```

but I'm getting `Unhandled Exception: 'package:easy_localization/src/easy_localization_app.dart': Failed assertion: line 274 pos 14: 'parent.supportedLocales.contains(locale)': is not true.`

As far as I can tell this is due to the fact that locale initialization has special logic for cutting off countryCode and falling back to default when locale is not found:
https://github.com/aissat/easy_localization/blob/661327d9a2338ac1342380381967b4021ebe773b/lib/src/easy_localization_app.dart#L163-L170
https://github.com/aissat/easy_localization/blob/661327d9a2338ac1342380381967b4021ebe773b/lib/src/easy_localization_app.dart#L152-L153

This logic is absent in locale setter or any other public function that I've found.

P.S.
It seems I could've avoided this problem by using `saveLocale` rather than having my own language settings. However in this case I don't see how to distinguish between “explicitly set the locale which happens to be equal to the current system default” vs “follow the system default”.

**Comment by Overman775** [comment]
startLocale is needed to skip determining the device locale, use saveLocale: true

default locale
```
import 'dart:io' show Platform;

String countryCode = Platform.localeName.split('_')[0];
String languageCode = Platform.localeName.split('_')[1];

context.locale = Locale(languageCode, countryCode); //set current device locale and refresh app
context.deleteSaveLocale(); //delete locale 
```

**Comment by amatveyakin** [comment]
As far as I can tell, this

```dart
String countryCode = Platform.localeName.split('_')[0];
String languageCode = Platform.localeName.split('_')[1];

context.locale = Locale(languageCode, countryCode);
```

is not a substitute for the startup locale detection logic. For example, this would throw if the device locale is not supported and a fallback is required. See more examples and links in my first comment.

The full logic is rather long: it includes the entire bodies of `_checkInitLocale`, `_getDeviceLocale`, `_getFallbackLocale` and parts of `_init` from https://github.com/aissat/easy_localization/blob/661327d9a2338ac1342380381967b4021ebe773b/lib/src/easy_localization_app.dart

This is more than a dozen lines of code. I can of course copy-paste them to my app, but in my opinion it's reasonable for the framework to provide such functionality out-of-the-box.

**Comment by Overman775** [comment]
@amatveyakin what are your suggestions?

**Comment by amatveyakin** [comment]
I think easy_localization should provide a method, say `resetLocale`, that encapsulates this behavior, i.e. “select the locale that would've been selected on start if `startLocale` were null and there were no saved locale”.

It should probably be combined with `deleteSaveLocale`. At least my intuition says you'd usually want to do both simultaneously.  To allow doing one without another, I would use optional arguments:
```dart
void resetLocale({
  bool resetCurrent = true,
  bool deleteSaved = true,
}) {
  ...
}
```

**Comment by Overman775** [comment]
@amatveyakin good idea, but need use only `delete Saved` = false argument in `setLocale`

```
void resetLocale({
  bool deleteSavedLocale = false,
}) {
 if(startLocale != null){
locale = startLocale;
}else{
locale = deviceLocale;
}  
 if(deleteSavedLocale){
  deleteSaveLocale()
 }
}
```
```
EasyLocalization.of(context).resetLocale();
//and contexte extension 
context.resetLocale();
```

**Comment by RaviKavaiya** [comment]
@Overman775 any updates on this?
This is a very much useful feature.

**Comment by cirnoftw** [comment]
As I can understand, there's currently no reliable way of resetting current locale with triggering widget rebuild?
I tried setting locale to platform locale and then calling `deleteSavedLocale`, but it has its own problems with not updating if platform locale matches with currently saved locale.

**Comment by Overman775** [comment]
PR #332 merget

**Comment by armandojimenez** [comment]
Can this be done? When I use resetLocale(), it does trigger a review and change, but it also saves the new language, so If I change the device language, it uses the one saved.
